flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -316,6 +316,14 @@ public StreamGraph generate() {
                 configuration.get(
                         ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH));
         shouldExecuteInBatchMode = shouldExecuteInBatchMode();
+        Optional<JobManagerOptions.SchedulerType> schedulerTypeOptional =
+                executionConfig.getSchedulerType();
+        boolean dynamic =
+                shouldExecuteInBatchMode
+                        && schedulerTypeOptional.orElse(
+                                        JobManagerOptions.SchedulerType.AdaptiveBatch)
+                                == JobManagerOptions.SchedulerType.AdaptiveBatch;
+        streamGraph.setDynamic(dynamic);
```

Comment:
How about to introduce a method `maybeSetDynamic()` and invoke it in `configureStreamGraph(...)`?

Reply:
Maybe add a hotfix commit to also move the above invocations `setAutoParallelismEnabled` and `setEnableCheckpointsAfterTasksFinish` into `configureStreamGraph()`.

Reply:
-> How about to introduce a method maybeSetDynamic() and invoke it in configureStreamGraph(...)?
Maybe `setDynamic(final StreamGraph graph)` ?